File: src/main/java/com/bookfair/entity/Reservation.java
package com.bookfair.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "reservations")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Reservation {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;
    
    @ManyToOne
    @JoinColumn(name = "stall_id", nullable = false)
    private Stall stall;
    
    @Column(nullable = false, unique = true)
    private String qrCode;
    
    @Column(nullable = false)
    private LocalDateTime createdAt = LocalDateTime.now();
}

File: frontend/src/shared/types/reservation.ts
import { User } from './user';
import { Stall } from './stall';

export interface Reservation {
    id: number
    user: User
    stall: Stall
    qrCode: string
    status: 'CONFIRMED' | 'CANCELLED'
    emailSent: boolean
    createdAt: string
}

export interface ReservationRequest {
    userId: number
    stallIds: number[]
}
